community-modules/core/src/ts/components/framework/userComponentFactory.ts

```
@@ -80,91 +81,91 @@ export class UserComponentFactory extends BeanStub {
     @Autowired('userComponentRegistry') private readonly userComponentRegistry: UserComponentRegistry;
     @Optional('frameworkComponentWrapper') private readonly frameworkComponentWrapper: FrameworkComponentWrapper;
 
-    public getHeaderCompDetails(colDef: ColDef, params: IHeaderParams): UserCompDetails | undefined {
+    public getHeaderCompDetails(colDef: ColDef, params: WithoutGridCommon<IHeaderParams>): UserCompDetails | undefined {
```

We use the `WithoutGridCommon` type so that we do not have to cast to interfaces which can lead to typing errors being hidden. It omits the common properties that will be attached centrally so that we do not have to worry about them where the params are created.